src/bsoncxx/oid.cpp

```diff
@@ -33,10 +33,11 @@ oid::oid(init_tag_t) : _is_valid(true) {
     std::memcpy(_bytes, oid.bytes, sizeof(oid.bytes));
 }
 
-oid::oid(stdx::string_view str) : _is_valid(bson_oid_is_valid(str.data(), str.length())) {
+oid::oid(string::view_or_value str)
```

Comment:
Seems like this could be `const string::view_or_value&`, since you aren't actually storing this anywhere, just copying it into the bson library.